src/network/core/packet.cpp

```diff
@@ -180,6 +180,15 @@ bool Packet::CanReadFromPacket(uint bytes_to_read)
 	return true;
 }
 
+/**
+ * Returns the number of bytes left to read to reach the end of the packet
+ * @return The amount of bytes left to be read
+ */
+int Packet::GetBytesLeftInPacket()
```

Comment:
Personally, I really wouldn't do this. It mostly adds complexity to the code that will never be used.

If we decide to change the protocol, that is an irreversible change. We upgrade the bananas-server first, merge this PR after, and that is kinda final for the protocol.

If we ever revert this functionality, it cannot revert the protocol, as we have clients that now depend on it. Patchpacks might already have copied it etc. So even if we revert, these bytes will remain reserved in the protocol till the end of time.

Say we revert and also remove the bytes from the protocol, and some time later someone adds another field to the protocol, older clients (and patchpacks that copied the functionality) will still read that new field as upload-date, which can only result in other problems. This is why not having a version field is so annoying btw :p

In other words: protocol changes are final and always only extending (never removing), and clients don't need code "in case the protocol is reverted". That is an impossible situation.

Again, at least, this is my opinion :) feel free to proof me wrong :D

Reply:
Agree on this. If you're going to extend the protocol, do it in an extensible way. For example use a tagged format like RIFF, which is also used for the savegame format (although only chunk-level is tagged there). Then the extended protocol would start with a "magic" tag, followed by number of bytes in the extended protocol part, followed by each add-on data element tagged.
But at that point you may as well replace the entire protocol.

Reply:
That is not what I said, so just to make sure there is no confusion about what I said:

I am perfectly fine with this PR, except for the part it keeps backwards compatibility with an old bananas-server. That is a scenario that simply is never going to happen, so I see no benefit of having code for that :) Just assuming this new field is always there, is more than sufficient in my opinion.
This mostly as I can fully imagine patchpacks releasing a version which include this PR, at which point reverting is impossible. So better already assume it is impossible, and build the code accordingly.

That is all I was saying :D Rewriting the full protocol should happen some day, but is, as far as I am concerned, out of scope for this PR.

Reply:
Thanks both. @nielsmh we discussed the extensibility of the data passing in https://github.com/OpenTTD/bananas-server/pull/43, and I think everyone would agree that we need to make a proper rewrite implementing this over JSON/proto or another more structured data messaging framework that also lets us pass versioning, etc.

But I can't do this now, it's too hard of a task and I mainly want to get this PR in so we can have this feature first. I honestly don't mind volunteering to get the long-term approach done, but that will require a proper design proposal and many approvals from the dev community.

Coming back to the topic: thanks @TrueBrain for clarifying. If you **really** believe I shouldn't worry about the possibility of having to revert BaNaNaS then I am fine removing it. Nicer, cleaner, and forward-looking, yeah!